src/pkcs11/pkcs11-spy.c

```diff
@@ -1430,52 +1454,7 @@ C_DeriveKey(CK_SESSION_HANDLE hSession, CK_MECHANISM_PTR pMechanism, CK_OBJECT_H
 
 	enter("C_DeriveKey");
 	spy_dump_ulong_in("hSession", hSession);
-	fprintf(spy_output, "[in] pMechanism->type=%s\n",
-		lookup_enum(MEC_T, pMechanism->mechanism));
-	switch (pMechanism->mechanism) {
-	case CKM_ECDH1_DERIVE:
-	case CKM_ECDH1_COFACTOR_DERIVE:
-		if (pMechanism->pParameter == NULL) {
-			fprintf(spy_output, "[in] pMechanism->pParameter = NULL\n");
-			break;
-		}
-		CK_ECDH1_DERIVE_PARAMS *param =
-			(CK_ECDH1_DERIVE_PARAMS *) pMechanism->pParameter;
-		fprintf(spy_output, "[in] pMechanism->pParameter = {\n\tkdf=%s\n",
-			lookup_enum(CKD_T, param->kdf));
-		fprintf(spy_output, "\tpSharedData[ulSharedDataLen] = ");
-		print_generic(spy_output, 0, param->pSharedData,
-			param->ulSharedDataLen, NULL);
-		fprintf(spy_output, "\tpPublicData[ulPublicDataLen] = ");
-		print_generic(spy_output, 0, param->pPublicData,
-			param->ulPublicDataLen, NULL);
-		fprintf(spy_output, "}\n");
-		break;
-	case CKM_ECMQV_DERIVE:
-		if (pMechanism->pParameter == NULL) {
-			fprintf(spy_output, "[in] pMechanism->pParameter = NULL\n");
-			break;
-		}
-		CK_ECMQV_DERIVE_PARAMS *param2 =
-			(CK_ECMQV_DERIVE_PARAMS *) pMechanism->pParameter;
-		fprintf(spy_output, "[in] pMechanism->pParameter = {\n\tkdf=%s\n",
-			lookup_enum(CKD_T, param2->kdf));
-		fprintf(spy_output, "\tpSharedData[ulSharedDataLen] =");
-		print_generic(spy_output, 0, param2->pSharedData,
-			param2->ulSharedDataLen, NULL);
-		fprintf(spy_output, "\tpPublicData[ulPublicDataLen] = ");
-		print_generic(spy_output, 0, param2->pPublicData,
-			param2->ulPublicDataLen, NULL);
-		fprintf(spy_output, "\tulPrivateDataLen = %lu",
-			param2->ulPrivateDataLen);
-		fprintf(spy_output, "\thPrivateData = %lu", param2->hPrivateData);
-		fprintf(spy_output, "\tpPublicData2[ulPublicDataLen2] = ");
-		print_generic(spy_output, 0, param2->pPublicData2,
-			param2->ulPublicDataLen2, NULL);
-		fprintf(spy_output, "\tpublicKey = %lu", param2->publicKey);
-		fprintf(spy_output, "}\n");
-		break;
-	}
+	spy_dump_mechanism("[in] pMechanism", pMechanism);
```

Comment:
I think the mechanism is always input parameter so the [in] should be implicit inside of the `spy_dump_mechanism()` to keep it in sync with the other uses below.

Reply:
OK